src/hal/user_comps/xhc-whb04b-6/hal.h

```diff
@@ -549,6 +549,10 @@ class Hal
     //! Writes the corresponding counter to to each axis' count.
     //! \param counters values to propagate to each axis
     void setJogCounts(const HandWheelCounters& counters);
+    //! waits until a given pin is set to a requested state 
```

Comment:
This comment is incorrect in two ways.

1. The function waits until the pin *no longer* equals the specified state, or
2. The function times out.

On return from this function there is no way to tell which end condition triggered the return.

Reply:
well I documented what this function should be used for.
Pin A will be changed, the feedback pin will report if it's acknowledged (that's the whole picture), of course if you only break out the function itself it will wait until a pin got changed.